components/shared/CurrencySelector.tsx

```diff
@@ -19,24 +20,38 @@ export const CurrencySelector: React.FC<CurrencySelectorProps> = ({
   hideDefaultCurrencies,
   ...props
 }) => {
-  const chainId = useSDKChainId();
-
+  const chainId = useSDKChainId() as number;
   const [isAddingCurrency, setIsAddingCurrency] = useState(false);
   const [editCustomCurrency, setEditCustomCurrency] = useState("");
   const [customCurrency, setCustomCurrency] = useState("");
   const [initialValue] = useState(value);
 
   const isCustomCurrency: boolean = useMemo(() => {
     if (initialValue && chainId && initialValue !== customCurrency) {
-      return !CURRENCIES[chainId]?.find(
-        (currency: CurrencyMetadata) =>
-          currency.address.toLowerCase() === initialValue.toLowerCase(),
-      );
+      if (chainId in CURRENCIES) {
+        return !CURRENCIES[chainId as SUPPORTED_CHAIN_ID]?.find(
+          (currency: CurrencyMetadata) =>
+            currency.address.toLowerCase() === initialValue.toLowerCase(),
+        );
+      }
+
+      // QUESTION - is this correct?
+      // for non-default chains
+      return true;
     }
```

Comment:
@jnsdls Need your input here. is this correct?

Reply:
This seems correct, yes